core/src/test/scala/unit/kafka/server/KafkaApisTest.scala

```diff
@@ -2822,6 +2822,31 @@ class KafkaApisTest extends Logging {
       () => kafkaApis.handleWriteTxnMarkersRequest(null, RequestLocal.withThreadConfinedCaching))
   }
 
+  @Test
+  def requiredAclsNotPresentWriteTxnMarkersThrowsAuthorizationException(): Unit = {
```

Comment:
For this test, it can pass without this change. Maybe we need a test to verify it won't throw exception when alter cluster is allowed, and clusterAction is denied, it won't throw exception. WDYT?

Reply:
That is the happy case path verified by existing tests. As mocks are not present there, by default the CLUSTER_ACTION check doesn't throw an exception, and the ALTER check returns false.

I can modify the existing tests to make that explicit.

Reply:
Did we make this modification?